frontend/taipy-gui/src/components/Taipy/Notification.tsx

```diff
@@ -0,0 +1,31 @@
+import React from "react";
+import Alert from "@mui/material/Alert";  
+import { TaipyBaseProps } from "./utils";
+import { useDynamicProperty } from "../../utils/hooks";
+
+interface NotificationProps extends TaipyBaseProps {
+    severity?: "error" | "warning" | "info" | "success";
+    message: string | (() => string);  // Dynamic string handling
+    variant?: "filled" | "outlined" | "standard";
+    defaultMessage?: string;
```

Comment:
Here is exactly where 'defaultMessage' would appear, and not on the back-end side.
That is relevant.

Reply:
a example of using dynamic property: 
```ts
interface CompProps {
  value?: string;
  defaultValue?: string;
}
```
In the body of you component:
```ts
const value = useDynamicProperty(props.value, props.defaultValue, "");
```

on the backend side
if value is default property
```py
.set_value_and_default() (var_type=dynamic_string is default)
```
else
```py
.set_attributes(
            [
                ("value", PropertyType.dynamic_string),
...
```